app/views/spree/orders/form/_update_buttons.html.haml

```diff
@@ -1,7 +1,7 @@
 .row
   .columns.small-12.medium-3
-    - if current_order.andand.distributor == @order.distributor
```

Comment:
Ok, this is not very pretty but I think I know what is happening.

`current_order` is the cart. `@order` is the order loaded by Sprees `OrdersController` to render the confirmation page. @order and current_order are (as far as I know) **NEVER** the same order, because as soon as an order is completed, a new `current_order` (for the same shop/distributor) is instantiated and loaded into the session to represent the cart. So in this case, the distributors of the `current_order` (cart) and the `@order` will be the same.

Users can navigate to the confirmation page for a particular order via their `/account` page, and in this case, the `current_order` could be for any shop on the site, so we don't want to show a 'back to shop' button that takes the user back to a completely unrelated shop, so it should be hidden in this instance.

Does that make sense?